src/actions/ui.js

```diff
@@ -19,6 +19,7 @@ import {
     UPDATE_UI_PROGRAM_ID,
     UPDATE_UI_PROGRAM_STAGE_ID,
     CLEAR_UI_PROGRAM,
```

Comment:
Are these 2 actually clearing the ids? Perhaps rename them to `CLEAR_UI_*_ID` as well.